src/nsrt_learning.py

```diff
@@ -34,52 +33,41 @@ def learn_nsrts_from_data(dataset: Dataset, predicates: Set[Predicate],
     segments = [seg for segs in segmented_trajs for seg in segs]
 
     # STEP 3: Cluster the data by effects, jointly producing one STRIPSOperator,
-    #         OptionSpec, and Partition (data store) per cluster. These items
-    #         then used to initialize _NSRTIntermediateData objects (NIDs).
+    #         NSRTDatastore, and OptionSpec per cluster. These items are then
+    #         used to initialize PartialNSRTAndDatastore objects (PNADs).
     #         Note: The OptionSpecs here are extracted directly from the data.
     #         If we are doing option learning, then the data will not contain
     #         options, and so the option_spec fields are just the specs of a
     #         DummyOption. We need a default dummy because future steps require
     #         the option_spec field to be populated, even if just with a dummy.
-    nids = _get_initial_nids(segments, verbose=CFG.do_option_learning)
+    pnads = _get_initial_pnads(segments, verbose=CFG.do_option_learning)
 
-    # STEP 4: Learn side predicates for the operators and update NIDs. These
+    # STEP 4: Learn side predicates for the operators and update PNADs. These
     #         are predicates whose truth value becomes unknown (for *any*
     #         grounding not explicitly in effects) upon operator application.
     if CFG.learn_side_predicates:
-        _learn_nid_side_predicates(nids)
+        _learn_pnad_side_predicates(pnads)
 
-    # STEP 5: Learn options (option_learning.py) and update NIDs.
-    _learn_nid_options(nids)
+    # STEP 5: Learn options (option_learning.py) and update PNADs.
+    _learn_pnad_options(pnads)
 
-    # STEP 6: Learn samplers (sampler_learning.py) and update NIDs.
-    _learn_nid_samplers(nids, do_sampler_learning)
+    # STEP 6: Learn samplers (sampler_learning.py) and update PNADs.
+    _learn_pnad_samplers(pnads, do_sampler_learning)
 
-    # STEP 7: Finalize and return the NSRTs.
-    nsrts = []
-    for nid in nids:
-        nid.finalize()
-        assert nid.nsrt is not None
-        nsrts.append(nid.nsrt)
+    # STEP 7: Print and return the NSRTs.
+    nsrts = [pnad.get_nsrt() for pnad in pnads]
     print("\nLearned NSRTs:")
     for nsrt in sorted(nsrts):
         print(nsrt)
     print()
     return set(nsrts)
 
 
-def learn_strips_operators(segments: Sequence[Segment], verbose: bool = True,
-                           ) -> Tuple[List[STRIPSOperator], List[Partition]]:
-    """A light wrapper around _get_initial_nids() that returns the operators
-    and partitions themselves, rather than exposing the internal data structure.
+def learn_strips_operators(segments: Sequence[Segment], verbose: bool = True
+                           ) -> List[PartialNSRTAndDatastore]:
+    """An alias for _get_initial_pnads.
```

Comment:
I'm confused, are you just renaming the function then? what's the purpose?

Reply:
fixed